### PR TITLE
CSCQVAIN-138: No automated acceptance/functional tests

### PR DESCRIPTION
**Pull request by jppiiroinen**
This adds initial tests and the structure for the tests.

### CODE REVIEW

**Comment by jppiiroinen** [comment]
changes are needed for the tests for Rights and Access tab which has been changed.

File: tests/fute/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+PYTHON_CMD:=source venv/bin/activate && source .secrets && python3
+
+all: venv check
+
+venv:
+	python3 -m venv venv
+	source venv/bin/activate && pip3 install https://github.com/CSCfi/tauhka/archive/master.zip
```
Comment (tahme) [inline tests/fute/Makefile]:
```
/bin/sh: 1: source: not found
```
Annoyingly, depending on the system, /bin/sh may or may not be a symlink to bash which supports source.

Maybe use . which is more portable.

Comment (jppiiroinen) [reply tests/fute/Makefile]: there is a bug, missing SHELL:=/bin/bash

Comment (jppiiroinen) [reply tests/fute/Makefile]: this is now fixed

File: tests/fute/Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	source venv/bin/activate && pip3 install -r requirements.txt
+
+chromedriver:
+	curl -O https://chromedriver.storage.googleapis.com/74.0.3729.6/chromedriver_mac64.zip
```
Comment (tahme) [inline tests/fute/Makefile]: I wonder if there's a good way to automatically install the correct version and keep it up to date with the installed Chrome version. Would it be possible to have a copy of a specific version of Chrome for testing?

Comment (jppiiroinen) [reply tests/fute/Makefile]: hmm.. there was something for that hmm..

Comment (jppiiroinen) [reply tests/fute/Makefile]: yeah there was some misc python projects, but all what they did is that they have the same url and some fixed value for the version. I rather see that this is a separate topic to be solved. at the moment there is 75 out there and it works with web driver for 74.

Comment (jppiiroinen) [reply tests/fute/Makefile]: it was noted in private chat, that there is API change between 75 and 74 WebDriver that will cause the tests to fail. To have support for 75 series an update to Tauhka is required.

File: src/widgets/Array.vue [hunk 3]
```diff
@@ -28,7 +29,7 @@
 						:path="newPath(index)"
 						:value="value[index]"
 						:parent="parent[property]"
-						:property="index"
+						:property="property + '_array_' + index"
```
Comment (tahme) [inline src/widgets/Array.vue]: This one's probably unintentional? (Or at least these break the component, see the keywords field for example.)

File: src/widgets/Array.vue [hunk 4]
```diff
@@ -50,15 +51,15 @@
...
```
Comment (tahme) [inline src/widgets/Array.vue]: This one's probably unintentional?

File: src/widgets/Array.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/widgets/Array.vue]: hmm

File: src/widgets/Array.vue (hunk 4 above)
Comment (jppiiroinen) [reply src/widgets/Array.vue]: true that..

File: src/widgets/Array.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/widgets/Array.vue]: fixed

File: src/widgets/Array.vue (hunk 4 above)
Comment (jppiiroinen) [reply src/widgets/Array.vue]: fixed

File: src/widgets/refdata/list-ui.vue [hunk 5]
```diff
@@ -5,7 +5,7 @@
     <b-form-group id="list-ui-form-group" class="list-ui-form-group" :class="isRequired ? 'required' : ''" :label-cols="uiLabel ? labelCols : 1" :description="uiDescription" :label="uiLabel">
       <b-input-group>
         <div v-if="type === 'multiselect'" class="flex-grow-1">
-          <Multiselect v-model="model" @input="setValue" :options="items" v-if="items" :customLabel="customLabel"
```
Comment (tahme) [inline src/widgets/refdata/list-ui.vue]: This gives an error when you open the files tab and have a file selected in the dataset. The list-ui component (used by FileItem.vue) does not have the `property` prop.

File: tests/fute/.gitignore [hunk 6]
```diff
@@ -0,0 +1,5 @@
+__pycache__
+venv
+chromedriver
+*.zip
+.secrets
```
Comment (tahme) [inline tests/fute/.gitignore]: missing newline at eof

File: tests/fute/qvaintestcase.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        self.open_usermenu()
+        usermenu_fullname = self.find_element("usermenu_fullname")
+        error_msg = "the user ({user} != {expected}) is not logged in"
+        was_user_fullname = os.environ["TEST_FULLNAME"] in usermenu_fullname.text
+        assert was_user_fullname, error_msg.format(user=usermenu_fullname.text, expected=os.environ["TEST_FULLNAME"])
+        self.close_usermenu()
+
+    def open_frontpage(self):
+        # The top does not actually get you to the frontpage
+        # appTopBarImage = self.driver.find_element_by_xpath('//*[@id="app-topbar"]/a/img')
```
Comment (tahme) [inline tests/fute/qvaintestcase.py]: I think you fixed this so the top image should now get you to the frontpage

Comment (jppiiroinen) [reply tests/fute/qvaintestcase.py]: true and that PR has landed already

File: tests/fute/.gitignore (hunk 6 above)
Comment (jppiiroinen) [reply tests/fute/.gitignore]: ack

File: src/widgets/refdata/list-ui.vue (hunk 5 above)
Comment (jppiiroinen) [reply src/widgets/refdata/list-ui.vue]: :/

**Comment by jppiiroinen** [comment]
```
$ make check
..
----------------------------------------------------------------------
Ran 2 tests in 102.956s

OK
```

File: src/widgets/refdata/list-ui.vue (hunk 5 above)
Comment (jppiiroinen) [reply src/widgets/refdata/list-ui.vue]: removed the change

File: tests/fute/.gitignore (hunk 6 above)
Comment (jppiiroinen) [reply tests/fute/.gitignore]: fixed

File: tests/fute/qvaintestcase.py (hunk 7 above)
Comment (jppiiroinen) [reply tests/fute/qvaintestcase.py]: fixed

**Review by tahme — APPROVED** [review]

**Pull request merged**